### PR TITLE
Add configuration option for base Python image for Docker files

### PR DESCRIPTION
**Pull request by alchzh**
[Python 3.8 is now officially EoL](https://discuss.python.org/t/python-3-8-is-now-officially-eol/66983). I've been testing how OFRAK does with newer python versions in the [`maintenance/newer-python-test-fixes`](https://github.com/redballoonsecurity/ofrak/tree/maintenance/newer-python-test-fixes) branch. This change just lets us build Docker images off images of newer Python versions.

- [x] I have reviewed the [OFRAK contributor guide](https://ofrak.com/docs/contributor-guide/getting-started.html) and attest that this pull request is in accordance with it.

**One sentence summary of this PR (This should go in the CHANGELOG!)**
Add configuration option for base Python image for Docker files.

**Link to Related Issue(s)**

**Please describe the changes in your request.**
- Adds a `python_image` configuration option to docker yaml files. Defaults to the current 3.8-bullseye image for now.

**Anyone you think should look at this, specifically?**

### CODE REVIEW

File: ofrak_core/Dockerstub [hunk 1]
```diff
@@ -61,8 +61,11 @@ RUN cd /tmp && \
 
 # Install Jefferson
 WORKDIR /tmp
-RUN wget https://bootstrap.pypa.io/pip/get-pip.py && python3.9 get-pip.py && python3.8 get-pip.py && rm get-pip.py
-RUN python3.9 -m pip install jefferson
+RUN wget https://bootstrap.pypa.io/pip/get-pip.py && \
+    /usr/bin/python3 get-pip.py && \
+    /usr/local/bin/python get-pip.py && \
+    rm get-pip.py
+RUN /usr/bin/python3 -m pip install jefferson
```
[the remark this answers is not in the document]
Comment (alchzh) [reply ofrak_core/Dockerstub:68]: Investigating this, looks like this is an artifact of #326 which added Jefferson back when the Docker image was based on python 3.7, which Jefferson didn't support. That pull request later bumped the Python version to 3.8 and added jefferson to our requirements.txt, so we end up having the `/usr/local/bin/python` (docker image) install of jefferson instead of the `/usr/bin/python` (debian) jefferson on our `PATH` anyway.

Also, `pip` always on the python version we actually use.

tl;dr this whole section is not needed.

**Comment by alchzh** [comment]
I ran into a problem where the new setuptools editable installs are broken with our directory layout. Directories in the working directory are now imported before editable installed packages, so running e.g. `import ofrak_type` from a folder with the parent `ofrak_type` directory is broken. See https://github.com/pypa/setuptools/issues/3557 and https://setuptools.pypa.io/en/latest/userguide/development_mode.html for details.

```
$ docker run -it redballoonsecurity/ofrak/dev-3.13 bash
Using service wrapper: yajsw-beta-13.01
Installing Ghidra Server...
Traceback (most recent call last):
  File "<frozen runpy>", line 189, in _run_module_as_main
  File "<frozen runpy>", line 148, in _get_module_details
  File "<frozen runpy>", line 112, in _get_module_details
  File "/ofrak_core/ofrak/__init__.py", line 8, in <module>
    from ofrak.ofrak_context import OFRAK, OFRAKContext
  File "/ofrak_core/ofrak/ofrak_context.py", line 13, in <module>
    from ofrak_type import InvalidStateError
ImportError: cannot import name 'InvalidStateError' from 'ofrak_type' (unknown location)
```

I see a couple solutions to this problem, not sure which one is the best
- add `--config-settings editable_mode=compat` to any `pip install -e` command
- Require older setuptools in pyproject.toml
```
[build-system]
requires = ["setuptools == 63"]
build-backend = "setuptools.build_meta"
```

**Comment by rbs-jacob** [comment]
> I ran into a problem where the new setuptools editable installs are broken with our directory layout. Directories in the working directory are now imported before editable installed packages, so running e.g. `import ofrak_type` from a folder with the parent `ofrak_type` directory is broken.

Is this only happening because the default working directory in the Docker images we build is the root directory? Would changing the `WORKDIR` to `/root/` as the last step in the Dockerfile assembled from Dockerstub files be an alternative way to address this?

**Review by rbs-jacob — APPROVED** [review]

**Comment by ANogin** [comment]
> I ran into a problem where the new setuptools editable installs are broken with our directory layout. Directories in the working directory are now imported before editable installed packages, so running e.g. `import ofrak_type` from a folder with the parent `ofrak_type` directory is broken. See [pypa/setuptools#3557](https://github.com/pypa/setuptools/issues/3557) and https://setuptools.pypa.io/en/latest/userguide/development_mode.html for details.
> 
> ```
> $ docker run -it redballoonsecurity/ofrak/dev-3.13 bash
> Using service wrapper: yajsw-beta-13.01
> Installing Ghidra Server...
> Traceback (most recent call last):
>   File "<frozen runpy>", line 189, in _run_module_as_main
>   File "<frozen runpy>", line 148, in _get_module_details
>   File "<frozen runpy>", line 112, in _get_module_details
>   File "/ofrak_core/ofrak/__init__.py", line 8, in <module>
>     from ofrak.ofrak_context import OFRAK, OFRAKContext
>   File "/ofrak_core/ofrak/ofrak_context.py", line 13, in <module>
>     from ofrak_type import InvalidStateError
> ImportError: cannot import name 'InvalidStateError' from 'ofrak_type' (unknown location)
> ```
> 
> I see a couple solutions to this problem, not sure which one is the best
> 
> * add `--config-settings editable_mode=compat` to any `pip install -e` command
> * Require older setuptools in pyproject.toml
> 
> ```
> [build-system]
> requires = ["setuptools == 63"]
> build-backend = "setuptools.build_meta"
> ```

@alchzh There is also https://github.com/python/mypy/issues/13392; I ended up using `export SETUPTOOLS_ENABLE_FEATURES="legacy-editable"` in [stale] https://github.com/redballoonsecurity/ofrak/pull/314/files#diff-9d7ea1b163b7b4693bbc13b0dd627154521dc56ad997cd3b1a7747e413916027
